### PR TITLE
Log enhancement: return error message when getting assignment servers and log exception when initializing

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
Log enhancement: return error message when getting assignment servers and log exception when initializing

### Why are the changes needed?
To be easy to find the cause of error.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
No need.

### CODE REVIEW

**Comment by jerqi** [comment]
Is this pr a compatible  improvement?

File: coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorServer.java [hunk 1]
```diff
@@ -49,8 +49,12 @@ public class CoordinatorServer {
 
   public CoordinatorServer(CoordinatorConf coordinatorConf) throws Exception {
     this.coordinatorConf = coordinatorConf;
-    initialization();
-
+    try {
```
[the remark this answers is not in the document]
Comment (zuston) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorServer.java:52]: I hope all initializing exception message can be recorded into log files. Sometimes due to abnormal configuration,  server or coordinator will exit directly no any log recorded, which is hard to debug.

**Comment by zuston** [comment]
> Is this pr a compatible improvement?

I think it's a compatible chang.

File: coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorServer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zuston) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorServer.java:52]: Yes. stdout message will lost when using ansible to deploy uniffle cluster.

**Comment by zuston** [comment]
Tested, it's OK @jerqi

**Review by jerqi — APPROVED** [review]
I verify the compatibility again. It's ok. LGTM. Thanks @zuston

**Pull request merged**